### PR TITLE
PyTorch documentation

### PR DESCRIPTION
**Pull request by rsarm**

### CODE REVIEW

**Comment by Danzelot** [comment]
It looks nice and I would like to see it on the docs.
@rsarm Is this still up to date? If not, what needs updating?

**Comment by rsarm** [comment]
No, it's not up to date. I will fix it.

File: docs/software/packages/pytorch.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# PyTorch on LUMI
+
+[PyTorch](https://pytorch.org) is an open source Python package that provides tensor computation, like NumPy, with GPU acceleration and deep neural networks built on a tape-based autograd system.
+
+PyTorch can be installed by the users following the code's [instructions](https://pytorch.org/get-started/locally/). The options to choose for LUMI in the interactive instructions are `Linux`, `Pip` and `ROCm5.X`. For installing with `pip`, the `cray-python` module should be loaded. PyTorch comes with ROCm binaries needed for the GPU support. Even if a particular version of ROCm is not avilable on LUMI, PyTorch may still be able to use the GPUs.
```
Comment (olouant) [inline docs/software/packages/pytorch.md]: typo

```suggestion
PyTorch can be installed by the users following the code's [instructions](https://pytorch.org/get-started/locally/). The options to choose for LUMI in the interactive instructions are `Linux`, `Pip` and `ROCm5.X`. For installing with `pip`, the `cray-python` module should be loaded. PyTorch comes with ROCm binaries needed for the GPU support. Even if a particular version of ROCm is not available on LUMI, PyTorch may still be able to use the GPUs.
```

File: docs/software/packages/pytorch.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+$> singularity exec -B $SCRATCH:$SCRATCH pytorch_rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1.sif bash
+Singularity> python -m venv pt_rocm5.4.1_env --system-site-packages
+Singularity> . pt_rocm5.4.1_env/bin/activate
+(pt_rocm5.4.1_env) Singularity> pip install python-hostlist
+```
+Now when running the container, the virtual environment must be activated before calling python.
+
+## Multi-GPU training
+
+The communication between LUMI's GPUs during training with Pytorch is done via [RCCL](https://github.com/ROCmSoftwarePlatform/rccl), which is a library of  collective communication routines for GPUs. RCCL works out of the box on LUMI's, however, a special plugin is required so it can take advantadge of the [Slingshot interconnect](https://www.hpe.com/emea_europe/en/compute/hpc/slingshot-interconnect.html). That's the [`aws-ofi-rccl`](https://github.com/ROCmSoftwarePlatform/aws-ofi-rccl) plugin, which is library that can be used as backend for RCCL to interact with the interconnect via libfabric.
```
Comment (olouant) [inline docs/software/packages/pytorch.md]: typo

```suggestion
The communication between LUMI's GPUs during training with Pytorch is done via [RCCL](https://github.com/ROCmSoftwarePlatform/rccl), which is a library of  collective communication routines for GPUs. RCCL works out of the box on LUMI's, however, a special plugin is required so it can take advantage of the [Slingshot interconnect](https://www.hpe.com/emea_europe/en/compute/hpc/slingshot-interconnect.html). That's the [`aws-ofi-rccl`](https://github.com/ROCmSoftwarePlatform/aws-ofi-rccl) plugin, which is a library that can be used as a back-end for RCCL to interact with the interconnect via libfabric.
```

File: docs/software/packages/pytorch.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# PyTorch on LUMI
+
+[PyTorch](https://pytorch.org) is an open source Python package that provides tensor computation, like NumPy, with GPU acceleration and deep neural networks built on a tape-based autograd system.
+
+PyTorch can be installed by the users following the code's [instructions](https://pytorch.org/get-started/locally/). The options to choose for LUMI in the interactive instructions are `Linux`, `Pip` and `ROCm5.X`. For installing with `pip`, the `cray-python` module should be loaded. PyTorch comes with ROCm binaries needed for the GPU support. Even if a particular version of ROCm is not avilable on LUMI, PyTorch may still be able to use the GPUs.
+
+PyTorch can be run within containers as well. In particular, containers from the images provided by [AMD on DockerHub](https://hub.docker.com/u/rocm).
+Those images are updated frequently and make possible to try PyTorch with recent ROCm versions.
```
Comment (olouant) [inline docs/software/packages/pytorch.md]:
```suggestion
Those images are updated frequently and make it possible to try PyTorch with recent ROCm versions. 
```

File: docs/software/packages/pytorch.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+export CXI_FORK_SAFE=1
+export CXI_FORK_SAFE_HP=1
+export FI_CXI_DISABLE_CQ_HUGETLB=1
+export SINGULARITYENV_LD_LIBRARY_PATH=/opt/ompi/lib:${EBROOTAWSMINOFIMINRCCL}/lib:/opt/cray/xpmem/2.4.4-2.3_9.1__gff0e1d9.shasta/lib64:${SINGULARITYENV_LD_LIBRARY_PATH}
+
+srun singularity exec -B"/appl:/appl" \
+                      -B"$SCRATCH:$SCRATCH" \
+                      $SCRATCH/pytorch_rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1.sif python cnn_distr.py
+```
+Here we have used a few environment variables. The ones starting with `NCCL_` and `CXI_`, as well as `FI_CXI_DISABLE_CQ_HUGETLB` are used by RCCL for the communication over Slingshopt. The `MIOPEN_` ones are needed to make [MIOpen](https://rocmsoftwareplatform.github.io/MIOpen/doc/html/index.html) create it's caches on `/tmp`. Finally, with `SINGULARITYENV_LD_LIBRARY_PATH` some directories are included in the container's `LD_LIBRARY_PATH`. This is important for RCCL to find the `aws-ofi-rccl` plugin. In addition, `NCCL_DEBUG=INFO`, can be used to increase RCCL's logging level to make sure that the `aws-ofi-rccl` plugin is being used: The lines
```
Comment (olouant) [inline docs/software/packages/pytorch.md]:
```suggestion
Here we have used a few environment variables. The ones starting with `NCCL_` and `CXI_`, as well as `FI_CXI_DISABLE_CQ_HUGETLB` are used by RCCL for the communication over Slingshopt. The `MIOPEN_` ones are needed to make [MIOpen](https://rocmsoftwareplatform.github.io/MIOpen/doc/html/index.html) create its caches on `/tmp`. Finally, with `SINGULARITYENV_LD_LIBRARY_PATH` some directories are included in the container's `LD_LIBRARY_PATH`. This is important for RCCL to find the `aws-ofi-rccl` plugin. In addition, `NCCL_DEBUG=INFO`, can be used to increase RCCL's logging level to make sure that the `aws-ofi-rccl` plugin is being used: The lines
```
